task_17/andrew.samsonov/src/main/java/ua/pp/darknsoft/CustomArrayList.java

```diff
@@ -0,0 +1,177 @@
+package ua.pp.darknsoft;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+public class CustomArrayList<T> implements CustomList<T> {
+
+  private T[] array;
+  private int size;
+  private static final int DEFAULT_CAPACITY = 10;
+
+  public CustomArrayList() {
+    array = (T[]) new Object[DEFAULT_CAPACITY];
+  }
+
+  public CustomArrayList(int cap) {
+    array = (T[]) new Object[cap];
+  }
+
+  @Override
+  public int size() {
+    return size;
+  }
+
+  @Override
+  public boolean isEmpty() {
+    return size == 0;
+  }
+
+  @Override
+  public boolean contains(T element) {
+    for (int i = 0; i < size; i++) {
+      if (array[i] == element) {
```

Comment:
What "==" will return if element is a Dog, provided we are comparing 2 different dog objects (but they are actually the same logically).